README.md
# Capernia Datapack: making the ultimate datapack for OTS (Open Tibia Server)[![Build Status](https://img.shields.io/badge/build-buggy-yellow.svg)]()


Capernia Datapack is a project born from the necessity of a serious OTS files.
Its goal is to make sure all new Tibia resources be unified and ready for use on the latest version of Forgotten Server.


![Tibia 10.91 Sample](https://cloud.githubusercontent.com/assets/19177793/15003885/258236d8-1187-11e6-86fd-f83118992146.png)


Capernia Datapack is made using **LUA scripting** language with some **mysql arguments**.

![Latest version!](https://img.shields.io/badge/tfs%20version-1.2-brightgreen.svg)
![There are newer versions!](https://img.shields.io/badge/client%20version-10.93-green.svg)
[![Gitter](https://badges.gitter.im/gordonbay/capernia-ots-datapack.svg)](https://gitter.im/gordonbay/capernia-ots-datapack?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)


Main Conceps
=============

* Analyse and scrap third party datapacks for new userful content
* Update the scripts making them compatible with latest version of Forgotten Server
* Unify all existing datapacks on Capernia datapack
* Help the community



Installation
============


On **Windows**, just extract all contents, change variables on **config.lua** to match your MYSQL settings, import MYSQL database (*.sql) and run the provided binary.

 - Unpack global.rar to same folder:

    > capernia-ots-datapack\data\world\global.rar

 - Edit config.lua settings:
![Lua script](https://img.shields.io/badge/script-lua-blue.svg)
	```lua
	mysqlHost = "127.0.0.1" 
	mysqlUser = "root"
	mysqlPass = "az88231015"
	mysqlDatabase = "servidor"
	```
 - Import `base.sql`to your mysql database:

    > capernia-ots-datapack\res\data.sql
    
 - Have fun!


Github Limitations
===========

Because of the `size` limits, the following file had to be compressed.

    capernia-ots-datapack\data\world\global.rar

Map Changes
===========

For obvious reasons we have a diferent protocol for changing the map.

 - Open a issue, using the `Issues` tab
 - Explain the changes needed and if possible, post an image

Datapacks Merges
=============

This list shows the status of third party datapacks **migrations**:

| Name | Last Author Release | Last Capernia Merge | Capernia Merge Status | Location | Report |
| --- | --- | --- | --- | --- | --- |
| Mitisuig | 13/04/2016 | 13/04/2016 | ![Complete!](https://img.shields.io/badge/status-complete-brightgreen.svg) | [TK](http://www.tibiaking.com/forum/topic/36403-global-full-tibiaking-10911092-tfs-12-full-oramond-cast-system-lions-rock-reward-system-equip-hotkey-new-items-news-monster-new-areas-krailos-asura-medusa-town-etc-eventos-sem-missao-v20/) | [Read Report](https://github.com/gordonbay/capernia-ots-datapack/issues/18) |
| Absolute | 03/05/2016 | - | ![Aware of this but nothing done yet!](https://img.shields.io/badge/aware-0%25-red.svg) | [TK](http://www.tibiaking.com/forum/topic/67075-global-full-download-1090-tfs-12ferumbras-questkrailosnew-arenacastrewardeventscasino/) | ![](https://img.shields.io/badge/invitation-pending-red.svg) |
| Lofiv | 02/03/2016 | - | ![Aware of this but nothing done yet!](https://img.shields.io/badge/aware-0%25-red.svg) | [OTLand](https://otland.net/threads/10-90-globalot-full-new-www-new-layout-krailos-tfs-1-2.241083/) | ![](https://img.shields.io/badge/invitation-pending-red.svg) |


Important Wiki Pages
=============

- [capernia-ots-datapack/wiki/Issues](https://github.com/gordonbay/capernia-ots-datapack/wiki/Issues) - Read this before report any issue


CHANGELOG.md
## 0.1.9 (07/06/2016)

Bugfixes:
  - fix problems with `weapons.xml` file as requested [here #40](https://github.com/gordonbay/capernia-ots-datapack/issues/40)


## 0.1.8 (04/06/2016)

Bugfixes:
  - fix wrong items values at `items.xml`
  - fix wrong textures and missing boat at `33358, 31588, 12` `33356, 31589, 11` `33365, 31612, 11` `33677, 31598, 7`
  - fix `Renegade Knight` drops and stats


## 0.1.7 (30/05/2016)

Bugfixes:
  - fix missing or wrong `items.xml` values that could make client crash. Hive structure, gooey substance, aniversary cake, stairs, adventurer stone, strange growth, insectoid cocoon, hive structure, insectoid hivecomb, dry insectoid hivecomb, insectoid eggs, cracked insectoid eggs, hive gate, cup of molten gold, tarantula trap, trapdoor, rolling pin, peeing orc, fire, destroy field rune, pitch black gap, hole, bog water, star ring, pallet, trunk, leaves, mountain, clay ground, the dead keeper, fire basin, table, dirt floor, rutted stone, jade ornaments, dragon claw lamp, iron case, putrid growth, acid, large black crystal, giant dreadcoil, doomato, witherstem, carnivortex, toxic tulip, clay lump, cursed shoulder spikes and others...
  - fix missing `schema.sql` file as requested [here #33](https://github.com/gordonbay/capernia-ots-datapack/issues/33)
  - fix `enchantment system` as requested [here #32](https://github.com/gordonbay/capernia-ots-datapack/issues/32)
  
  Enhancements:
  - fix wrong NPC `Yaman, Haroun and Grizzly Adams` coordinates as requested [here #32](https://github.com/gordonbay/capernia-ots-datapack/issues/32)

  Additions:
  - `Gesior` php data is now supported by this version

## 0.1.6 (27/05/2016)

Bugfixes:
  - fix `The Explorer Society Quest` errors as requested [here #27](https://github.com/gordonbay/capernia-ots-datapack/issues/27) 

## 0.1.5 (25/05/2016)

Bugfixes:
  - fix `spellbook` errors as requested [here #25](https://github.com/gordonbay/capernia-ots-datapack/issues/25) 

## 0.1.4 (23/05/2016)

Enhancements:
  - fix some `map` glitches
  - some changes on `Killing in the Name of...` necromancer task as partially requested [here #22](https://github.com/gordonbay/capernia-ots-datapack/issues/22) 
  - changed the compiled executables, now it comes both x64 and x86 versions and with respective dll

## 0.1.3 (20/05/2016)

Bugfixes:
  - fix `The New Frontier Quest` teleport not checking player storage as requested [here #21](https://github.com/gordonbay/capernia-ots-datapack/issues/21) 

## 0.1.2 (19/05/2016)

Bugfixes:
  - fix `surprise bags` errors as requested [here #20](https://github.com/gordonbay/capernia-ots-datapack/issues/20) 

## 0.1.1 (18/05/2016)

Enhancements:
  - fix wrong `spells` formulas as requested [here #19](https://github.com/gordonbay/capernia-ots-datapack/issues/19) 

## 0.1.0 (17/05/2016)

Bugfixes:
  - fix missing `monsters spells`

## 0.0.9 (17/05/2016)

Bugfixes:
  - fix wrong monsters `corpses` as requested [here #17](https://github.com/gordonbay/capernia-ots-datapack/issues/17)
  - fix missing `Yalahar` gates and `Captain Max` not checking mission status in order to teleport a character. As described [here #16](https://github.com/gordonbay/capernia-ots-datapack/issues/16)
  - fix Edron's `Bog Raider's` missing spaws and wrong map textures as requested [here #14](https://github.com/gordonbay/capernia-ots-datapack/issues/14)

## 0.0.8 (14/05/2016)

Enhancements:
  - added `10.92` new itens and potions

## 0.0.7 (13/05/2016)

Bugfixes:
  - fix a bug on `Ferumbras' Ascendant Quest` as requested [here #13](https://github.com/gordonbay/capernia-ots-datapack/issues/13)

Enhancements:
  - `Reward Chest` has been removed from Thais due violations of Capernia rules about custom content


## 0.0.6 (12/05/2016)

Bugfixes:
  - fix errors on NPC `Chemar`
  - fix incorrect textures on `Darashia` -9 level, at rotworms cave
  
Enhancements:
  - NPC Chemar now respect `The New Frontier` quest in order to take you to `Farmine`


## 0.0.5 (11/05/2016)

Bugfixes:
  - fix melee weapons not being able to cut webs on the `Cruzader Helmet Quest`
  
Enhancements:
  - melee weapons now break or destroy some itens
  - spider eggs should work properly now

## 0.0.4 (10/05/2016)

Additions:
  - added missing `spaws` around theese spots:
    - 33351,31706,13
    - 33351,31678,13
    - 33351,31671,13
    - 33330,31692,12
    - 33337,31650,12
    - 33351,31671,13
    - 32987,31629,12
    - 33190,32056,11
    - 33190,32042,11
    - 33183,32035,11
    - 33162,32000,11
    - 33204,31986,11
    - 33365,31692,11
    - 33379,31671,11
    - 33414,31650,11
    - 33197,32434,9
    - 33211,32441,9
    - 33204,32413,9
    - 33337,31993,9
    - 33337,31979,9
    - 33001,31678,9
    - 33008,31650,9
    - 33576,31599,9
    - 32652,33007,8
    - 32593,32994,8
    - 32659,32983,8
    - 32619,32978,8
    - 32154,32868,8
    - 33330,31979,8
    - 33316,31965,8
    - 33092,31713,8
    - 32154,32868,7
    - 32854,32798,7
    - 33176,32469,7
    - 32154,32861,6
    - 33505,31608,7
  
Enhancements:
  - added an entire missing island underground at 33188,32988,7
  - fixed glitchs on the map file
  - removed Thais trainers because of the new Capernia rules on custom stuff

## 0.0.3 (09/05/2016)

Additions:
  - added missing NPC `Ulala`
  - added missing NPC `A Dark Priestess`
  - added missing NPC `Rapanaio`
  - added missing NPC `Mother Of Jack`
  - added missing NPC `Pythius The Rotten`
  - added missing NPC `Awarness Of The Emperor`
  - added missing NPC `A Ghostly Woman`
  - added missing NPC `Junkar`
  - added missing NPC `The Queen Of The Banshees`
  - added missing NPC `Grombur`
  - added missing NPC `Frafnar`
  - added missing NPC `Nokmir`
  - added missing NPC `Robson`
  - added missing NPC `A Wrinkled Bonelord`
  - added missing NPC `Shortsighted Dwarf`
  - added missing NPC `A Tortured Soul`
  - added missing NPC `A Lost Soul`
  - added missing NPC `Zirkon`
  - added missing NPC `Skip`
  - added missing NPC `Polly`
  - added missing NPC `Asrak`

data/npc/scripts/Captain Max.lua
local keywordHandler = KeywordHandler:new()
local npcHandler = NpcHandler:new(keywordHandler)
NpcSystem.parseParameters(npcHandler)

function onCreatureAppear(cid)			npcHandler:onCreatureAppear(cid)			end
function onCreatureDisappear(cid)		npcHandler:onCreatureDisappear(cid)			end
function onCreatureSay(cid, type, msg)		npcHandler:onCreatureSay(cid, type, msg)		end
function onThink()		npcHandler:onThink()		end

local voices = { {text = 'Whoah. That was a large shadow passing by.'} }
npcHandler:addModule(VoiceModule:new(voices))


local travelNode = keywordHandler:addKeyword({'calassa'}, StdModule.say, {npcHandler = npcHandler, text = 'That is quite a long unprofitable travel. I\'ll bring you to Calassa though. Do you want me to do it?'}, function(player) return player:getStorageValue(Storage.ExplorerSociety.QuestLine) >= 50 end)
	travelNode:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = false, level = 0, cost = 200, destination = Position(31911, 32710, 6) })
	travelNode:addChildKeyword({'no'}, StdModule.say, {npcHandler = npcHandler, reset = true, text = 'Maybe another time, then.'})

local travelNode = keywordHandler:addKeyword({'yalahar'}, StdModule.say, {npcHandler = npcHandler, text = 'That is quite a long unprofitable travel. I\'ll bring you to Yalahar. Is that ok with you?'}, function(player) return player:getStorageValue(Storage.TheWayToYalahar.QuestLine) >= 1 end)
	travelNode:addChildKeyword({'yes'}, StdModule.travel, {npcHandler = npcHandler, premium = true, level = 0, cost = 400, destination = Position(32816, 31272, 6) })
	travelNode:addChildKeyword({'no'}, StdModule.say, {npcHandler = npcHandler, reset = true, text = 'We would like to serve you some time.'})

keywordHandler:addKeyword({'sail'}, StdModule.say, {npcHandler = npcHandler, text = 'Welcome on board, noble |PLAYERNAME|. I can bring you to {Calassa} or {Yalahar}, but only if you have the according mission from {Berenice} or {Wyrdin}.'})
keywordHandler:addKeyword({'passage'}, StdModule.say, {npcHandler = npcHandler, text = 'Welcome on board, noble |PLAYERNAME|. I can bring you to {Calassa} or {Yalahar}, but only if you have the according mission from {Berenice} or {Wyrdin}.'})
keywordHandler:addKeyword({'job'}, StdModule.say, {npcHandler = npcHandler, text = 'I am the captain of this ship.'})
keywordHandler:addKeyword({'captain'}, StdModule.say, {npcHandler = npcHandler, text = 'I am the captain of this ship.'})

npcHandler:setMessage(MESSAGE_GREET, "Ahoy, |PLAYERNAME|. On a mission for the explorer society, eh?")
npcHandler:setMessage(MESSAGE_FAREWELL, "Good bye.")
npcHandler:setMessage(MESSAGE_WALKAWAY, "Good bye.")
npcHandler:addModule(FocusModule:new())


data/actions/scripts/other/potions.lua
local config = {
	-- strong health potion
	[7588] = {health = {min = 250, max = 350}, vocations = {3, 4}, text = 'paladins and knights', level = 50, emptyId = 7634},
	-- strong mana potion
	[7589] = {mana = {min = 115, max = 185}, vocations = {1, 2, 3}, text = 'sorcerers, druids and paladins', level = 50, emptyId = 7634},
	-- great mana potion
	[7590] = {mana = {min = 150, max = 250}, vocations = {1, 2}, text = 'sorcerers and druids', level = 80, emptyId = 7635},
	-- Ultimate Mana Potion
	[26029] = {mana = {min = 400, max = 600}, vocations = {1, 2}, text = 'sorcerers and druids', level = 130, emptyId = 7635},
	-- great health potion
	[7591] = {health = {min = 425, max = 575}, vocations = {4}, text = 'knights', level = 80, emptyId = 7635},
	-- health potion
	[7618] = {health = {min = 125, max = 175}, emptyId = 7636},
	-- mana potion
	[7620] = {mana = {min = 75, max = 125}, emptyId = 7636},
	-- great spirit potion
	[8472] = {health = {min = 250, max = 350}, mana = {min = 100, max = 200}, vocations = {3}, text = 'paladins', level = 80, emptyId = 7635},
	-- Ultimate Spirit Potion
	[26030] = {health = {min = 400, max = 550}, mana = {min = 150, max = 250}, vocations = {3}, text = 'paladins', level = 130, emptyId = 7635},
	-- ultimate health potion
	[8473] = {health = {min = 650, max = 850}, vocations = {4}, text = 'knights', level = 130, emptyId = 7635},
	-- supreme health potion
	[26031] = {health = {min = 800, max = 1000}, vocations = {4}, text = 'knights', level = 200, emptyId = 7635},
	-- antidote potion
	[8474] = {antidote = true, emptyId = 7636},
	-- small health potion
	[8704] = {health = {min = 60, max = 85}, emptyId = 7636}
}

local antidote = Combat()
antidote:setParameter(COMBAT_PARAM_TYPE, COMBAT_HEALING)
antidote:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_MAGIC_BLUE)
antidote:setParameter(COMBAT_PARAM_TARGETCASTERORTOPMOST, true)
antidote:setParameter(COMBAT_PARAM_AGGRESSIVE, false)
antidote:setParameter(COMBAT_PARAM_DISPEL, CONDITION_POISON)

local exhaust = Condition(CONDITION_EXHAUST_HEAL)
exhaust:setParameter(CONDITION_PARAM_TICKS, (configManager.getNumber(configKeys.EX_ACTIONS_DELAY_INTERVAL) - 100))
-- 1000 - 100 due to exact condition timing. -100 doesn't hurt us, and players don't have reminding ~50ms exhaustion.

function onUse(player, item, fromPosition, target, toPosition, isHotkey)
	local potion = config[item.itemid]
	if not potion then
		return true
	end

	if target.itemid ~= 1 or target.type ~= THING_TYPE_PLAYER then
		return false
	end




	if player:getCondition(CONDITION_EXHAUST_HEAL) then
		player:sendTextMessage(MESSAGE_STATUS_SMALL, Game.getReturnMessage(RETURNVALUE_YOUAREEXHAUSTED))
		return true
	end

	if potion.antidote and not antidote:execute(target, Variant(target.uid)) then
		return false
	end

	if (potion.level and player:getLevel() < potion.level)
			or (type(potion.vocations) == 'table' and not isInArray(potion.vocations, player:getVocation():getBase():getId()))
			and not (player:getGroup():getId() >= 2) then
		player:say(string.format('This potion can only be consumed by %s of level %d or higher.', potion.text, potion.level), TALKTYPE_MONSTER_SAY)
		return true
	end

	if type(potion.health) == 'table' and not doTargetCombatHealth(0, target, COMBAT_HEALING, potion.health.min, potion.health.max, CONST_ME_MAGIC_BLUE) then
		return false
	end

	if type(potion.mana) == 'table' and not doTargetCombatMana(0, target, potion.mana.min, potion.mana.max, CONST_ME_MAGIC_BLUE) then
		return false
	end

	player:addAchievementProgress('Potion Addict', 100000)

	player:addCondition(exhaust)
	doCreatureSayWithRadius(target, 'Aaaah...', TALKTYPE_MONSTER_SAY, 2, 2)

	local topParent = item:getTopParent()
	if topParent.isItem and (not topParent:isItem() or topParent.itemid ~= 460) then
		local parent = item:getParent()
		if not parent:isTile() and (parent:addItem(potion.emptyId, 1) or topParent:addItem(potion.emptyId, 1)) then
			item:remove(1)
			return true
		end
	end

	Game.createItem(potion.emptyId, 1, item:getPosition())
	item:remove(1)
	return true
end


config.lua
-- Combat settings
-- NOTE: valid values for worldType are: "pvp", "no-pvp" and "pvp-enforced"
worldType = "pvp"
hotkeyAimbotEnabled = true
protectionLevel = 70
killsToRedSkull = 6
killsToBlackSkull = 9
pzLocked = 60000
removeChargesFromRunes = false
timeToDecreaseFrags = 24 * 60 * 60 * 1000
whiteSkullTime = 15 * 60 * 1000
stairJumpExhaustion = 1500
experienceByKillingPlayers = true
expFromPlayersLevelRange = 200

-- Connection Config
-- NOTE: maxPlayers set to 0 means no limit
ip = "127.0.0.1"
bindOnlyGlobalAddress = false
loginProtocolPort = 7171
gameProtocolPort = 7172
statusProtocolPort = 7171
maxPlayers = 1000
motd = "Acessem nosso site Sejam bem vindos"
onePlayerOnlinePerAccount = true
allowClones = false
serverName = "TESTSERVER"
statusTimeout = 5000
replaceKickOnLogin = true
maxPacketsPerSecond = 25

enableLiveCasting = true
liveCastPort = 7173

-- Deaths
-- NOTE: Leave deathLosePercent as -1 if you want to use the default
-- death penalty formula. For the old formula, set it to 10. For
-- no skill/experience loss, set it to 0.
deathLosePercent = -1

-- Houses
-- NOTE: set housePriceEachSQM to -1 to disable the ingame buy house functionality
housePriceEachSQM = 2000
houseRentPeriod = "weekly"

-- Item Usage
timeBetweenActions = 200
timeBetweenExActions = 1000

-- Map
-- NOTE: set mapName WITHOUT .otbm at the end
mapName = "global"
mapAuthor = "eumesmo"

-- Market
marketOfferDuration = 7 * 24 * 60 * 60
premiumToCreateMarketOffer = false
checkExpiredMarketOffersEachMinutes = 60
maxMarketOffersAtATimePerPlayer = 100

-- MySQL
mysqlHost = "127.0.0.1"
mysqlUser = "root"
mysqlPass = ""
mysqlDatabase = "ot"
mysqlPort = 3306
mysqlSock = ""
passwordType = "sha1"

-- Misc.
allowChangeOutfit = true
freePremium = true
kickIdlePlayerAfterMinutes = 30
maxMessageBuffer = 4
emoteSpells = false
classicEquipmentSlots = false

-- Rates
-- NOTE: rateExp is not used if you have enabled stages in data/XML/stages.xml
rateExp = 400
rateSkill = 60
rateLoot = 5
rateMagic = 10
rateSpawn = 6

-- Monsters
deSpawnRange = 2
deSpawnRadius = 50

-- Stamina
staminaSystem = true

-- Scripts
warnUnsafeScripts = true
convertUnsafeScripts = true

-- Startup
-- NOTE: defaultPriority only works on Windows and sets process
-- priority, valid values are: "normal", "above-normal", "high"
defaultPriority = "high"
startupDatabaseOptimization = false

-- Status server information
ownerName = "Legend of Chaos"
ownerEmail = "yves@sigmarhohewb.com.br"
url = "http://legendofchaos.com.br"
location = "Brazil"